lib/rollbax/logger.ex

```diff
@@ -69,7 +69,7 @@ defmodule Rollbax.Logger do
   end
 
   def handle_event({level, _gl, event}, %{metadata: keys} = state) do
-    if proceed?(event) and meet_level?(level, state.level) do
+    if proceed?(event) and meet_level?(level, state.level) and !whitelisted?(state[:whitelist], event) do
```

Comment:
Actually, in this implementation we're building the same `message` with `IO.chardata_to_string/1` *twice*. What about this implementation:

```elixir
if proceed?(even) and meet_level?(level, state.level) do
  post_event_unless_in_blacklist(state, level, event, keys)
end
{:ok, state}
```

and then:

```elixir
defp post_event_unless_in_blacklist(state, level, {Logger, message, event_time, meta}, keys) do
  message = message |> prune_chardata() |> IO.chardata_to_string()
  unless Enum.any?(state[:blacklist], &Regex.match?(&1, message)) do
     # old stuff ending in Rollbax.Client.emit/5
  end
end
```